src/_pytest/python.py

```diff
@@ -195,7 +195,11 @@ def pytest_pycollect_makeitem(collector, name, obj):
     if res is not None:
         return
     # nothing was collected elsewhere, let's do it here
-    if isclass(obj):
+    try:
```

Comment:
if this was factored into a function called `_hardened_isclass` the test could be a unittest

Reply:
I've factored it out and created a unit test, but kept the integration test, since it not only tests if `_safe_isclass` is really used during collection, but would also show other issues in the future with Django's settings.